src/core.js

```diff
@@ -141,26 +146,34 @@ define([
             csrftoken: _this.opts().csrftoken
         });
 
-        var hasBrokenReferences = _.debounce(function () {
-            return _this.data.core.form.hasBrokenReferences();
+        var validateForSave = _.debounce(function () {
+            var form = _this.data.core.form;
+            if (form.hasBrokenReferences()) {
+                return {
+                    title: "Errors in Form",
+                    content: "<div class='alert alert-danger'>Form has " +
+                        "reference errors.<br>Look for questions marked with " +
+                        "<i class='fd-valid-alert-icon fa fa-warning'></i> " +
+                        "and check they don't reference deleted questions.</div>",
+                };
+            } else if (!form.isFormValid(validateMug)) {
```

Comment:
Was thinking about this in bed last night, and I remembered why I didn't do this when I originally implemented the save button warning. This traverses all mugs in the form, so for very large forms could introduce a pause on mouse over the save button. I have some ideas of how to improve it:

- Validate current mug since that is likely the one that needs it most (easy)
- Check if there are other validation errors in the form in some way that is not O(n). I can't remember if that's possible right now or if we need to implement it? (maybe harder)